program/rust/src/accounts/price.rs

```diff
@@ -382,115 +295,23 @@ impl PriceFeedMessage {
-            .read_i64::<BigEndian>()
-            .map_err(|_| OracleError::DeserializationError)?;
-        let ema_price = cursor
-            .read_i64::<BigEndian>()
-            .map_err(|_| OracleError::DeserializationError)?;
-        let ema_conf = cursor
-            .read_u64::<BigEndian>()
-            .map_err(|_| OracleError::DeserializationError)?;
-
-
-        Ok(Self {
-            id,
-            price,
-            conf,
-            exponent,
-            publish_time,
-            prev_publish_time,
-            ema_price,
-            ema_conf,
-        })
+        bytes.to_vec()
     }
 }
 
-/// Message format for sending Twap data via the accumulator program
-#[repr(C)]
-#[derive(Debug, Copy, Clone, PartialEq)]
-#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-pub struct TwapMessage {
-    pub id:                [u8; 32],
-    pub cumulative_price:  i128,
-    pub cumulative_conf:   u128,
-    pub num_down_slots:    u64,
-    pub exponent:          i32,
-    pub publish_time:      i64,
-    pub prev_publish_time: i64,
-    pub publish_slot:      u64,
-}
-
-#[allow(dead_code)]
-impl TwapMessage {
-    // The size of the serialized message. Note that this is not the same as the size of the struct
-    // (because of the discriminator & struct padding/alignment).
-    pub const MESSAGE_SIZE: usize = 1 + 32 + 16 + 16 + 8 + 4 + 8 + 8 + 8;
-    pub const DISCRIMINATOR: u8 = 1;
-
-    pub fn from_price_account(key: &Pubkey, account: &PriceAccountV2) -> Self {
-        let publish_time = if account.agg_.status_ == PC_STATUS_TRADING {
-            account.timestamp_
-        } else {
-            account.prev_timestamp_
-        };
-
-        Self {
-            id: key.to_bytes(),
-            cumulative_price: account.price_cumulative.price,
-            cumulative_conf: account.price_cumulative.conf,
-            num_down_slots: account.price_cumulative.num_down_slots,
-            exponent: account.exponent,
-            publish_time,
-            prev_publish_time: account.prev_timestamp_,
-            publish_slot: account.last_slot_,
-        }
-    }
-
-    /// Serialize this message as an array of bytes (including the discriminator)
-    /// Note that it would be more idiomatic to return a `Vec`, but that approach adds
-    /// to the size of the compiled binary (which is already close to the size limit).
+impl PythOracleSerialize for TwapMessage {
+    /// Ideally, this structure should not be aware of the discrminator and the Message enum
+    /// should handle serializing the discriminator. However, this method is kept as is in order
+    /// to avoid any further increase in the program binary size since we are close to the limit.
     #[allow(unused_assignments)]
-    pub fn to_bytes(self) -> [u8; Self::MESSAGE_SIZE] {
-        let mut bytes = [0u8; Self::MESSAGE_SIZE];
+    fn to_bytes(self) -> Vec<u8> {
+        const MESSAGE_SIZE: usize = 1 + 32 + 16 + 16 + 8 + 4 + 8 + 8 + 8;
```

Comment:
I discussed this with Ali, we were thinking of moving the DISCRIMINATOR serialization part to the Message enum and implement to_bytes for it as well as the next step. This would lead to `PythOracleSerialize` becoming more abstract and not having a fixed `discriminator`/`message_size`. That's why we kept it like this.